src/sentry/integrations/slack/tasks.py

```diff
@@ -170,6 +180,33 @@ def find_channel_id_for_alert_rule(organization_id, uuid, data, alert_rule_id=No
             redis_rule_status.set_value("failed")
             return
 
+    # This function call will write "input_channel_id" to the appropriate actions before we create the serializer
+    # I wasn't sure of a way to use the same loop to return True in the endpoints and to also do the lookup and didn't want to duplicate the loop
+    try:
+        mapped_ids = get_slack_channel_ids(organization, user, data)
+    except (serializers.ValidationError, ChannelLookupTimeoutError, InvalidTriggerActionError):
+        # channel doesn't exist error or validation error
+        logger.info(
+            "get_slack_channel_ids.failed",
+            extra={
+                "user_id": user.id if user is not None else None,
+                "organization_id": organization_id,
+                "data": data,
+            },
+        )
+        redis_rule_status.set_value("failed")
+        return
+
+    for trigger in data["triggers"]:
+        for action in trigger["actions"]:
+            if action["targetIdentifier"] in mapped_ids:
```

Comment:
@iProgramStuff Maybe we should check the `action["type"]` before checking if it's a `mapped_ids`? While it's highly unlikely, it is possible someone could make another alert rule action that has the same `targetIdentifier` but for a different integration (say MS Teams).